src/features/movies/moviesSaga.js

```diff
@@ -14,8 +17,19 @@ function* fetchPopularMoviesHandler() {
     yield call(alert, "Coś poszło nie tak! Spróbuj ponownie później.");
     yield put(fetchPopularMoviesError());
   }
-}
+};
+
+function* fetchGenresHandler() {
+  try {
+    const popularMovies = yield call(getGenres);
+    yield put(fetchGenresSuccess(popularMovies));
+  } catch (error) {
+    yield call(alert, "Coś poszło nie tak! Spróbuj ponownie później.");
+    yield put(fetchGenresError());
+  }
+};
 
 export function* watchFetchPopularMovies() {
   yield takeLatest(fetchPopularMovies.type, fetchPopularMoviesHandler);
-}
+  yield takeLatest(fetchGenres.type, fetchGenresHandler);
+};
```

Comment:
czy to nie powinno być w funkcji (function* watchFetchGenres) ? czy może ta poprzednia funkcja jest niepotrzebnie stworzona ?

Reply:
właśnie tutaj się inspirowałem to-do-listą, bo tam wrzucaliśmy wszystkie takeLatest do jednej funkcji watch i działało :P